scrapy/core/downloader/handlers/http11.py

```diff
@@ -445,6 +447,17 @@ def __init__(self, finished, txresponse, request, maxsize, warnsize, fail_on_dat
         self._fail_on_dataloss_warned = False
         self._reached_warnsize = False
         self._bytes_received = 0
+        self._ip_address = None
+
+    def connectionMade(self):
+        if self._ip_address is None:
+            peer = self.transport._producer.getPeer()
+            if isinstance(peer, TxIPv4Address):
+                self._ip_address = IPv4Address(type=peer.type, host=peer.host, port=peer.port)
+            # FIXME: uncomment once IPv6 is supported
```

Comment:
This could either be kept like this, or be removed and added when IPv6 support is added. The same with the `scrapy.utils.datatypes.IPv6Address` definition.